src/basics/Basics.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace basics;
+
+class Basics implements BasicsInterface
+{
+    protected $validator;
+
+    public function __construct($validator)

```

Comment:
Please change this to 
`public function __construct(private BasicValidatorInterface $validator) {
}`